24_clustering.Rmd

```diff
@@ -17,9 +18,122 @@ features/samples that share pattern.  For example, with clustering, we
 can find group of samples that share similar community
 composition. There are multiple clustering algorithms available.
 
+## Introduction to bluster package
+
+*bluster* is a Bioconductor package specialized in clustering. It offers 
+multiple algorithms such as hierarchical clustering, DBSCAN, K-means, 
+amongst others. The first thing to do when using this package is to load 
+it. It's also possible to use `transformCounts` to transform the data 
+for the clustering.
+```{r load-bluster}
+# Load dependencies
+if (!require(bluster)) {
+  install.packages("bluster")
+  library(bluster)
+}
+
+# Optional: Apply transformation
+tse <- transformCounts(tse, method = "relabundance")
+```
+
+The clustering can be done on the features or the samples.
+
+### Samples clustering
+
+To cluster the samples, we simply need to transpose the assay so that 
+the samples are in rows.
+```{r sample-clust-prep}
+assay <- t(assay(tse, "relabundance"))
+```
+
+Then, to do the clustering, use the `clusterRows` function. Depending 
+on its parameters, it will perform a different algorithm. Here, we'll 
+perform a hierarchical clustering.
+```{r bluster-sample-hclust1}
+# Simple use of the hierarchical clustering which sets the cut height to
+# half the dendrogram height.
+hclust.out <- clusterRows(assay, HclustParam())
```

Comment:
you can also just insert here directly the thing you defined above.

t(assay(tse, "relabundance"))

Reply:
Since I use the variable `x` twice, I think it's better to leave it that way to avoid doing the operation twice.